docs/manual/java/releases/Migration16.md

```diff
@@ -111,3 +113,31 @@ This is a summary of changes in Lagom 1.6 that would require a full cluster shut
 * The change in [[Akka Remote|Migration16#Remoting-Artery]] default implementation.
 * The change in default [[Shard Coordination|Migration16#Shard-Coordination]] strategy.
 * The change in [[Cassandra plugin version|Migration16#Akka-Persistence-Cassandra-Update]]. Only impact Lagom applications using Cassandra.
+
+### Multi-step upgrade: serialization
+
+Changes in default serialisers require rollingupgrades to happen in two steps.
+
+#### Step 1: new binary, old serialisers
+
+First, you will have to deploy a version of your binary that's already a Lagom 1.6.x application but you must set it up to use the serialisers that are compatible with Lagom 1.5.x so you old nodes can continu to operate. On you `application.conf` use:
+
+```
+## Lagom 1.6 no longer enables Java serialization by default but some 
+## code in Lagom 1.5 (or your own code) still used it so we enable it back.
+akka.actor.allow-java-serialization = on
```

Comment:
Is that true? I'm surprised that we had internal code using java serialization.

Reply:
I didn't track it down. It's most likely user code on the test app I used and the logs didn't point to any class in particular. I checked the code and everything looked fine.

I just checked again and noticed [`Get`](https://github.com/renatocaval/lagom-1.6-playground/blob/1351fc2561d7bf59074e9f17c79c3ffc9bb3c1af/shopping-cart-java/node1/shopping-cart/src/main/java/com/example/shoppingcart/impl/ShoppingCartCommand.java#L46) so maybe that was it.

In any case, Lagom or user code, with this trick you are safe (as long as the offending class is not an event serializing data in Java into the journal). Alternatively, users would have to do the serializer dance (from Java to JSON in a Lagom 1-5-based version of their code) and once all the deployed nodes use JSON only, _then_ start migrating to Lagom 1.6.